heidrun/digcomm_mods.rb

```diff
@@ -0,0 +1,289 @@
+
+    # <mods:physicalDescription><mods:extent>
+    extent record.field('mods:physicalDescription', 'mods:extent')
+
+    # <mods:genre...>
+    # FIXME: non-DCMIType values from type will be handled in enrichment
+    dcformat record.field('mods:genre')
+
+    identifier record.map(&build_identifier).flatten
+
+    # FIXME: Language code enrichment needed here (language_to_lexvo.rb)
+    #
+    # <mods:language><mods:languageTerm type="text" authority="iso639-2b"
+    # authorityURI="http://id.loc.gov/vocabulary/iso639-2"
+    # valueURI="http://id.loc.gov/vocabulary/iso639-2/code [value]">[text value]
+    language record.field('mods:language', 'mods:languageTerm')
+              .match_attribute(:type, 'text')
+              .match_attribute(:authority, 'iso639-2b')
+              .match_attribute(:authorityURI, 'http://id.loc.gov/vocabulary/iso639-2')
+
+    # FIXME: hierarchicalGeographic is currently coming out as a string like:
+    #
+    #  "\n      North and Central America\n      United States\n      Massachusetts\n      Essex\n      Marblehead\n    "
+    #
+    # Do we want to turn this into a comma-separated string instead?  The
+    # original ingest broke the hierarchicalGeographic into its component fields
+    # (city, county, state, ...) but MAPV4 doesn't have these fields anymore.
+    #
+    # Might also want to run the `split_coordinates.rb` enrichment to split
+    # lat/longs where they appear.
+    #
+    # <mods:subject><mods:hierarchicalGeographic>
+    # <mods:subject><mods:geographic>
+    # <mods:subject><mods:cartographics><mods:coordinates>
+    spatial :class => DPLA::MAP::Place,
+            :each => record.fields(['mods:subject', 'mods:hierarchicalGeographic'],
+                                   ['mods:subject', 'mods:geographic'],
+                                   ['mods:subject',
+                                    'mods:cartographics',
+                                    'mods:coordinates']),
+            :as => :place do
+      providedLabel place.map(&:value)
+    end
+
+    # <mods:originInfo><mods:place><mods:placeTerm type="text">
+    # <mods:originInfo><mods:publisher>
+    publisher :class => DPLA::MAP::Agent,
+              :each => record.fields(['mods:originInfo', 'mods:placeTerm'],
+                                     ['mods:originInfo', 'mods:publisher']),
+              :as => :publisher do
+      providedLabel publisher
+    end
+
+    relation record.map(&build_relation).flatten
+
+    # <mods:accessCondition...>
+    rights record.field('mods:accessCondition')
+
+    # Any <mods:subject> field, except <mods:hierarchicalGeographic>,
+    # <mods:geographic>, and <mods:cartographics><mods:coordinates>. [LCSH
+    # subfield values are separated with hyphenation "--"] Some will be repeats
+    # of place, but that's OK.
+    subject :class => DPLA::MAP::Concept,
+            :each => record.field('mods:subject')
+                    .reject { |v| v.child?('mods:hierarchicalGeographic') }
```

Comment:
This looks good. Like the suggested place revision in https://github.com/dpla/heidrun-mappings/pull/38/files#r48180898, it suggests some DSL improvements around `#child?` filters for `ValueArray`. :+1: on this serial `#reject` for now.